lib/matplotlib/axes/_base.py

```diff
@@ -113,9 +116,14 @@ def _process_plot_format(fmt):
                 raise ValueError(
                     'Illegal format string "%s"; two color symbols' % fmt)
             color = c
+        elif c == 'C' and i < len(chars) - 1:
+            color_cycle_number = int(chars[i + 1])
```

Comment:
Yes, we should probably document that limitation.  The problem with opening it up to more than 10 is that it makes backward compatibility much harder.  Should `C11` be the 12th color or the 2nd color with a triangle up marker?  If we did allow more than one digit, we'd force people to have to write `1C1`.

Since more than 10 colors is sort of an edge case, and ill-advised even then, I think it's fine to have this limitation.